tests/e2e/snapshot_vmservice_vm.go

```diff
@@ -0,0 +1,3434 @@
+
+	f := framework.NewDefaultFramework("vmsvc")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+	f.SkipNamespaceCreation = true // tests will create their own namespaces
+	var (
+		client                     clientset.Interface
+		namespace                  string
+		datastoreURL               string
+		storagePolicyName          string
+		storageClassName           string
+		storageProfileId           string
+		vcRestSessionId            string
+		vmi                        string
+		vmClass                    string
+		vmopC                      ctlrclient.Client
+		cnsopC                     ctlrclient.Client
+		isVsanHealthServiceStopped bool
+		isSPSserviceStopped        bool
+		vcAddress                  string
+		restConfig                 *restclient.Config
+		snapc                      *snapclient.Clientset
+		pandoraSyncWaitTime        int
+		err                        error
+		dsRef                      types.ManagedObjectReference
+		labelsMap                  map[string]string
+	)
+
+	ginkgo.BeforeEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// client connection
+		client = f.ClientSet
+		bootstrap()
+
+		// fetch the testbed type for executing testcases
+		topologyFeature := os.Getenv(topologyFeature)
+
+		// fetching nodes and reading storage policy name
+		if topologyFeature != topologyTkgHaName && topologyFeature != podVMOnStretchedSupervisor {
+			nodeList, err := fnodes.GetReadySchedulableNodes(ctx, f.ClientSet)
+			framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+			if !(len(nodeList.Items) > 0) {
+				framework.Failf("Unable to find ready and schedulable Node")
+			}
+			storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		} else {
+			storagePolicyName = GetAndExpectStringEnvVar(envZonalStoragePolicyName)
+		}
+
+		// fetching vc ip and creating creating vc session
+		vcAddress = e2eVSphere.Config.Global.VCenterHostname + ":" + sshdPort
+		vcRestSessionId = createVcSession4RestApis(ctx)
+
+		// reading storage class name for wcp setup "wcpglobal_storage_profile"
+		storageClassName = strings.ReplaceAll(storagePolicyName, "_", "-") // since this is a wcp setup
+
+		// fetching shared datastore url
+		datastoreURL = GetAndExpectStringEnvVar(envSharedDatastoreURL)
+
+		// reading datastore morf reference
+		dsRef = getDsMoRefFromURL(ctx, datastoreURL)
+		framework.Logf("dsmoId: %v", dsRef.Value)
+
+		// reading storage profile id of "wcpglobal_storage_profile"
+		storageProfileId = e2eVSphere.GetSpbmPolicyID(storagePolicyName)
+
+		/* creating/reading content library
+		   "https://wp-content-pstg.broadcom.com/vmsvc/lib.json" */
+		contentLibId := createAndOrGetContentlibId4Url(vcRestSessionId, GetAndExpectStringEnvVar(envContentLibraryUrl),
+			dsRef.Value, GetAndExpectStringEnvVar(envContentLibraryUrlSslThumbprint))
+
+		/*
+		   [ ~ ]# kubectl get vmclass -n csi-vmsvcns-2227
```

Comment:
remove comments

Reply:
No, I have intentionally left these comments because the VM service feature is new to everyone. The comments provide context on what each code block is doing and how to verify it on the backend. This helps the end user understand the end-to-end VM service code flow. 

I'm leaving this comment open because I want to understand why you're asking me to remove it.

Reply:
I have removed all the description comments.